packages/lib/payment/handlePayment.ts

```diff
@@ -0,0 +1,53 @@
+import appStore from "@calcom/app-store";
+import { EventTypeAppsList } from "@calcom/app-store/utils";
+import { EventTypeModel } from "@calcom/prisma/zod";
+import { CalendarEvent } from "@calcom/types/Calendar";
+
+import { AppCategories, Prisma } from ".prisma/client";
+
+const handlePayment = async (
+  evt: CalendarEvent,
+  selectedEventType: Pick<Zod.infer<typeof EventTypeModel>, "metadata">,
+  paymentAppCredentials: {
+    key: Prisma.JsonValue;
+    appId: EventTypeAppsList;
+    app: {
+      dirName: string;
+      categories: AppCategories[];
+    } | null;
+  },
+  booking: {
+    user: { email: string | null; name: string | null; timeZone: string } | null;
+    id: number;
+    startTime: { toISOString: () => string };
+    uid: string;
+  }
+) => {
+  const paymentApp = appStore[paymentAppCredentials?.app?.dirName as keyof typeof appStore];
+  if (!(paymentApp && "lib" in paymentApp && "PaymentService" in paymentApp.lib)) {
+    console.warn(`payment App service of type ${paymentApp} is not implemented`);
+    return null;
+  }
+  const PaymentService = paymentApp.lib.PaymentService;
+  const paymentInstance = new PaymentService(paymentAppCredentials);
+  const paymentData = await paymentInstance.create(
```

Comment:
This should create payment on provider and payment on our end with provider data.